### PR TITLE
The client now accepts args a a host and port

### PR DESCRIPTION
**Pull request by mprtcz**
The client now accepts args as ip address and port number, when absent it gets a default ones which are localhost and port 6666;

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 1]
```diff
@@ -94,27 +87,51 @@ private void initializeElementsAfterServerConnection() {
      */
     @FXML
     public void onFeatureButtonClicked() {
-        this.acceptANameAndStartANewGame(this.nameTextField.getText());
+        this.acceptANameAndStartNewGame(this.nameTextField.getText());
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: Is this keyword neccessary?

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java [hunk 2]
```diff
@@ -30,6 +30,7 @@ public void proceed() {
             try {
                 Thread.sleep(100);
             } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java]: 🍺 for this

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java (hunk 1 above)
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 3]
```diff
@@ -94,27 +87,51 @@ private void initializeElementsAfterServerConnection() {
...
     }
 
     @FXML
     public void onNameTextFieldEntered() {
-        this.acceptANameAndStartANewGame(this.nameTextField.getText());
+        this.acceptANameAndStartNewGame(this.nameTextField.getText());
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍 / 2

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java [hunk 4]
```diff
@@ -30,6 +30,7 @@ public void proceed() {
             try {
                 Thread.sleep(100);
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipGame.java]: :+1:

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 5]
```diff
@@ -145,10 +163,21 @@ void onSendToServerButtonClicked() {
...
-            game.runTheGame();
-        } catch (InterruptedException e) {
-            logger.log(DEFAULT_LEVEL, "Game interrupted");
-        }}).start();
+        new Thread(() -> {
+            try {
+                game.runTheGame();
+            } catch (InterruptedException e) {
+                logger.log(DEFAULT_LEVEL, "Game interrupted");
+                Thread.currentThread().interrupt();
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

File: src/main/java/com/spanish_inquisition/battleship/server/ClientConnectionHandler.java [hunk 6]
```diff
@@ -99,7 +101,8 @@ private void getMessageFromUser(BufferedReader input) {
             if(input != null) {
```
Comment (Devas) [inline src/main/java/com/spanish_inquisition/battleship/server/ClientConnectionHandler.java]: Too much ifs

File: src/test/java/com/spanish_inquisition/battleship/client/MainMenuControllerTest.java [hunk 7]
```diff
@@ -38,7 +39,7 @@ public void testOnFeatureButtonClicked() throws Exception {
         mainMenuController.nameTextField.setText("DummyText");
         assertTrue(mainMenuController.playerNameVBox.isVisible());
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/MainMenuControllerTest.java]: Use SoftAssert if you have multiple assertions

File: src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java [hunk 8]
```diff
@@ -56,36 +57,28 @@
...
     @FXML
     Button fleetSetupButton;
     @FXML
     BorderPane mainBorderPane;
+    @FXML
+    ProgressIndicator socketProgressIndicator;
 
     SocketClient socketClient;
     Game game;
+    private int port = 6666;
```
Comment (Dzionek95) [inline src/main/java/com/spanish_inquisition/battleship/client/MainMenuController.java]: 👍

**Pull request merged**